packages/grafana-ui/src/components/BigValue/BigValueLayout.tsx

```diff
@@ -103,16 +103,23 @@ export abstract class BigValueLayout {
     return styles;
   }
 
-  getPercentChangeStyles(percentChange: number): PercentChangeStyles {
+  // TODO update this to incorporate percent change color mode
+  getPercentChangeStyles(
+    percentChange: number,
+    percentChangeColorMode: PercentChangeColorMode | undefined,
+    valueStyles: React.CSSProperties
+  ): PercentChangeStyles {
     const VALUE_TO_PERCENT_CHANGE_RATIO = 2.5;
     const valueContainerStyles = this.getValueAndTitleContainerStyles();
     const percentFontSize = Math.max(this.valueFontSize / VALUE_TO_PERCENT_CHANGE_RATIO, 12);
     let iconSize = Math.max(this.valueFontSize / 3, 10);
 
     const color =
-      percentChange > 0
-        ? this.props.theme.visualization.getColorByName('green')
-        : this.props.theme.visualization.getColorByName('red');
+      percentChangeColorMode === PercentChangeColorMode.SameAsValue
```

Comment:
this might be good to just break out into a separate function to clean up this logic a bit as the amount of `?` and `:` is maybe too high haha

Reply:
with this broken out as a separate function adding a basic unit test would be pretty easy (basically to confirm that a given option is returning the expected color)